src/sentry/utils/openai_sdk_integration.py

```diff
@@ -0,0 +1,75 @@
+import hashlib
+
+from sentry_sdk.hub import Hub
+from sentry_sdk.integrations import DidNotEnable, Integration
+
+try:
+    from openai import resources
+except ImportError:
+    raise DidNotEnable("OpenAI is not installed")
+
+
+class OpenAiIntegration(Integration):
+    identifier = "openai"
+
+    def __init__(self, capture_prompts=False):
+        self.capture_prompts = capture_prompts
+
+    @staticmethod
+    def setup_once():
+        patch_openai()
+
+
+def patch_openai():
+    old_open_ai_resources_chat_completions_create = resources.chat.completions.Completions.create
+
+    def monkeypatched_openai_completions_create(*args, **kwargs):
+        hub = Hub.current
+
+        integration = hub.get_integration(OpenAiIntegration)
+        if integration is None:
+            return old_open_ai_resources_chat_completions_create(*args, **kwargs)
+
+        # for now, simply make the description (which is how grouping is defined)
+        # simply the first message in the chat. This is not ideal, but it's a start.
+        with hub.start_span(
+            op="ai.llm.completion",
+            description=_get_description(kwargs.get("messages"), integration.capture_prompts),
+        ) as span:
+            if integration.capture_prompts:
+                span.set_data(
+                    "chat_completion_input",
+                    {"messages": kwargs.get("messages")},
+                )
+            return_value = old_open_ai_resources_chat_completions_create(*args, **kwargs)
+            if getattr(return_value, "usage", None):
+                completion_output = {
+                    "created": return_value.created,
+                    "id": return_value.id,
+                }
+                if integration.capture_prompts:
```

Comment:
would it make more sense to make the output capturing a separate option than `capture_prompts`, like `capture_response`?

Reply:
this is a good question. It seems like to me if you have PII in the prompt, you'd certainly risk it being in the response as well. what do you think?

Reply:
Makes sense, maybe just a simple rename this option to `capture_data` or something more universal to make it clear that this triggers capturing the response too not just the prompt?